tests/Unit/Core/Foundation/VersionTest.php

```diff
@@ -442,10 +442,10 @@ public function provideVersions()
             '7.6.0' => [
                 '7.6.0',
                 [
-                    'version' => '1.7.6.0',
-                    'fullVersion' => '1.7.6.0',
+                    'version' => '7.6.0',
```

Comment:
Interesting.

We could say this test is _wrong_ because you changed the expected result.

But 7.6.0 indeed matches SemVer and not legacy style. Previous implementation would match it to legacy style although it made no sense.

So we have an issue
- do we change this behavior because it's better this way?
- do we keep the behavior of before because we're afraid of changing this that might be used by modules (so we keep wrong behavior of the past to provide backward compatibility)

Reply:
I agree with the idea that `7.6.*` are correct semver and should have been handled correctly historically Though I have the same doubts regarding changing this behaviour I'm not sure about the impact this all could have

Previously major versions were not indicated in the first digit, that's a weird way of handling versions but maybe we should keep this in our history of code as well. So my suggestion would be not to change this behaviour, and the Version class should handle the fact that for versions prior to the 8 (so up until the 7) the version must be written with a prefix `1.`

Which means that in the PrestaShop version management `1.7.5` is equal to `7.5`, equal in the sense that those two inputs will identify the same version which will actually be transformed back or converted to `1.7.5` in all cases.

Reply:
So I think it would mean adding a condition in the Version class:

```php
$isLegacy = $major < 8;
```

Reply:
Maybe the legacy pattern check is not the right one :thinking:

Reply:
Maybe ...


We can mix the 2 ways to determinate if it's a legacy version :

1 - If I have a 4 digits version 1.7.6.0 for example. Major = 7, minor = 6, patch = 0
2 - If I have a major version lower than 8.<br>7.6.0 will be equal to 1.7.6.0. Major = 7, minor = 6, patch = 0

Reply:
Maybe we could uniformize this:
- split the input version into 4 parts (when available use 0 as fallback for missing parts)
- if first part is `1` it is the legacy writing so we need to offset the index for each part
  - major is `$matches[2]`
  - minor is `$matches[3]`
  - patch is `$matches[4]`
- for other cases it means the input version respects the usual semver notation so we don't need to ignore the first match:
  - major is `$matches[1]`
  - minor is `$matches[2]`
  - patch is `$matches[3]`
- as for the output (aka fullVersion aka $this->version) as long as the major is < 8 we always display with the legacy style `1.{major}.{minor}.{patch}` (it is not correct semver but that was our legacy way 🤷‍♂️ )

Reply:
An alternative could also be to have two different getters, one that handles this switch between legacy/semver and another one which always respect semver (so it would return `7.5` for a `1.7.5` input)

Reply:
(Oh actually we already have this with `getSemVersion()` my bad)

Reply:
The suggestion of @jolelievre seems very good to me 👍